packages/firebase_ai/firebase_ai/lib/src/api.dart

```diff
@@ -1020,6 +1022,17 @@ final class GenerationConfig extends BaseGenerationConfig {
   ///   a schema; currently this is limited to `application/json`.
   final Schema? responseSchema;
 
+  /// The response schema as a JSON string.
+  ///
+  /// - Note: This only applies when the [responseMimeType] supports a schema;
+  ///   currently this is limited to `application/json`.
+  ///
+  /// This schema can include more advanced features of JSON than the [Schema]
+  /// class supports.  See [the Gemini
+  /// documentation](https://ai.google.dev/gemini-api/docs/structured-output#json-schema)
+  /// about the limitations of this feature.
+  final String? responseJsonSchema;
```

Comment:
what's the purpose of using a String as the type, instead of for json object we usually can do Map<String, Object?>?

Reply:
I was wrong about it wanting a string: it expects a JS object anyhow.